testing/expected-m.go

```diff
@@ -173,3 +173,26 @@ func CheckDate(expectedStr string, format string) func(i interface{}) error {
 		return errors.New(fmt.Sprintf("Expected date %s but got %s", expected, found))
 	}
 }
+
+func CheckDateClose(target time.Time, leeway time.Duration) func(interface{}) error {
```

Comment:
Lacks documentation about what this function is for and why it's just a wrapper for a func.

Reply:
I am against the dual functionality of this method. It is verbose and lacks aim. You should reduce this to 2 methods:
A) TimeInLeeway - Takes a time and duration - returns a bool if it satisfies
B) ParseTimeCheckLeeway - Takes in a time, duration, and format - returns a bool and error (utilizing TimeInLeeway)
You also do not need to return a method.

Reply:
ExpectedM key:value pairs can have either variables as the value or functions. If it's a value it will use `reflect` to compare the value against the data that the ExpectedM object is being compared against. If it's a function, it will pass the value of the data into the function and check whether that function returns an error.

CheckDateClose generates one of those functions.

I added documentation to explain this better, and create a new type for these checker functions.

I put those time functions into our `time` package in golib so that aren't directly in our testing/expected_m package.